### ISSUE TITLE
cosine learning rate

### ISSUE THREAD

**Issue by ChunmianLin** [Byronnar/tensorflow-serving-yolov3#4]
你好，我想请问下你在学习YunYang这份代码时，有没有注意到其中learning rate设置。这里用的是cosine learning+warmup的方法，但我看得觉得很奇怪。找出了相关论文对照了下公式，感觉对应不上。特别是代码中global_step的设置很奇怪，不知道你是否注意到了。
本来我应该去问原作者的，但他经常不回复或者回复延迟，所以只好来请教你。

**Comment by Byronnar** [comment]
我也注意到了这个学习率的问题，是有些奇怪，后面我准备换一种试试看，感觉这个学习率不能很直观的看到变化，还容易出现过拟合。到时候一起研究一下啊。

**Comment by ChunmianLin** [comment]
我今天换了，按照余弦学习率的公式替换进去，结果没感觉效果，而且还出现loss nan。现在不知道怎么搞，想问你下出现nan，你一般怎么调？我自己改了模型结构，想重新训练，所以现在nan问题很难解决。学习率我也调整了，调大调小似乎都没用。发自我的iPhone------------------ Original ------------------From: Byronnar  <notifications@github.com>Date: Wed,Oct 30,2019 9:38 AMTo: Byronnar/tensorflow-serving-yolov3 <tensorflow-serving-yolov3@noreply.github.com>Cc: ChunmianLin <cmlin@buaa.edu.cn>, Author <author@noreply.github.com>Subject: Re: [Byronnar/tensorflow-serving-yolov3] cosine learning rate (#4)我也注意到了这个学习率的问题，是有些奇怪，后面我准备换一种试试看，感觉这个学习率不能很直观的看到变化，还容易出现过拟合。到时候一起研究一下啊。



—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.

[

{

"@context": "http://schema.org",

"@type": "EmailMessage",

"potentialAction": {

"@type": "ViewAction",

"target": "https://github.com/Byronnar/tensorflow-serving-yolov3/issues/4?email_source=notifications\u0026email_token=AMWSBR762NASKDZO7NAT26DQRDQSBA5CNFSM4JGHJ6N2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOECSUFZI#issuecomment-547701477",

"url": "https://github.com/Byronnar/tensorflow-serving-yolov3/issues/4?email_source=notifications\u0026email_token=AMWSBR762NASKDZO7NAT26DQRDQSBA5CNFSM4JGHJ6N2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOECSUFZI#issuecomment-547701477",

"name": "View Issue"

},

"description": "View this Issue on GitHub",

"publisher": {

"@type": "Organization",

"name": "GitHub",

"url": "https://github.com"

}

}

]

**Comment by ChunmianLin** [comment]
数据没有改变，和没改网络结构之前是一样的，而之前倒不会一直出现nan。出现nan时调小学习率一般也就正常了。这次改了结构，怎么调学习率都没用。结构的话，我也是基于3的基础调节了一些模块，没有大的改动，也检查了几遍的输入输出，不知道怎么会这样。发自我的iPhone------------------ Original ------------------From: Byronnar  <notifications@github.com>Date: Wed,Oct 30,2019 5:34 PMTo: Byronnar/tensorflow-serving-yolov3 <tensorflow-serving-yolov3@noreply.github.com>Cc: ChunmianLin <cmlin@buaa.edu.cn>, Author <author@noreply.github.com>Subject: Re: [Byronnar/tensorflow-serving-yolov3] cosine learning rate (#4)

我今天换了，按照余弦学习率的公式替换进去，结果没感觉效果，而且还出现loss nan。现在不知道怎么搞，想问你下出现nan，你一般怎么调？我自己改了模型结构，想重新训练，所以现在nan问题很难解决。学习率我也调整了，调大调小似乎都没用。发自我的iPhone------------------ Original ------------------From: Byronnar notifications@github.comDate: Wed,Oct 30,2019 9:38 AMTo: Byronnar/tensorflow-serving-yolov3 tensorflow-serving-yolov3@noreply.github.comCc: ChunmianLin cmlin@buaa.edu.cn, Author author@noreply.github.comSubject: Re: [Byronnar/tensorflow-serving-yolov3] cosine learning rate (#4)我也注意到了这个学习率的问题，是有些奇怪，后面我准备换一种试试看，感觉这个学习率不能很直观的看到变化，还容易出现过拟合。到时候一起研究一下啊。 —You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe. [ { "@context": "http://schema.org", "@type": "EmailMessage", "potentialAction": { "@type": "ViewAction", "target": "#4?email_source=notifications\u0026email_token=AMWSBR762NASKDZO7NAT26DQRDQSBA5CNFSM4JGHJ6N2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOECSUFZI#issuecomment-547701477", "url": "#4?email_source=notifications\u0026email_token=AMWSBR762NASKDZO7NAT26DQRDQSBA5CNFSM4JGHJ6N2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOECSUFZI#issuecomment-547701477", "name": "View Issue" }, "description": "View this Issue on GitHub", "publisher": { "@type": "Organization", "name": "GitHub", "url": "https://github.com" } } ]



nan 问题，情况有点多，数据方面，如果有脏数据会出现，然后学习率不能太高，再就是 网络结构问题了。 得根据你改的那个算法来了。



—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or unsubscribe.

[

{

"@context": "http://schema.org",

"@type": "EmailMessage",

"potentialAction": {

"@type": "ViewAction",

"target": "https://github.com/Byronnar/tensorflow-serving-yolov3/issues/4?email_source=notifications\u0026email_token=AMWSBR6U6GJUWEOEXO47CLTQRFIIZA5CNFSM4JGHJ6N2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOECTPV4A#issuecomment-547814128",

"url": "https://github.com/Byronnar/tensorflow-serving-yolov3/issues/4?email_source=notifications\u0026email_token=AMWSBR6U6GJUWEOEXO47CLTQRFIIZA5CNFSM4JGHJ6N2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOECTPV4A#issuecomment-547814128",

"name": "View Issue"

},

"description": "View this Issue on GitHub",

"publisher": {

"@type": "Organization",

"name": "GitHub",

"url": "https://github.com"

}

}

]

**Comment by Byronnar** [comment]
是的，转换成pb模型进行调用。

**Comment by ChunmianLin** [comment]
谢谢。你最近有时间调试模型啥的吗，有时间多交流下？抠细节的时候，还是有些地方看得迷糊，理解起来有问题。

**Comment by Byronnar** [comment]
> 谢谢。你最近有时间调试模型啥的吗，有时间多交流下？抠细节的时候，还是有些地方看得迷糊，理解起来有问题。

最近可能有点忙，不过你可以留一个联系方式，我加你，有时间讨论。

**Comment by ChunmianLin** [comment]
行啊，你加我qq：846901365。有时间就交流交流，随时都可以，不一定非要等到专门有空的时间。
